Some integration-cli tests pull busybox from Docker Hub

Description of problem:

Running the `make test-integration-cli` on ARM shows some errors. It seems that the test **TestDockerNetworkHostModeUngracefulDaemonRestart** and maybe some others don't use the pinned and preloaded container image `busybox`. Instead it pulls it directly from Docker Hub.

For ARM and other architectures this then results in an `exec format error` as the Linux Intel image is pulled and started on ARM.

`docker version`:
```
Client:
 Version:      1.9.0
 API version:  1.21
 Go version:   go1.4.3
 Git commit:   76d6bc9
 Built:        Tue Nov  3 22:16:18 UTC 2015
 OS/Arch:      linux/arm

Server:
 Version:      1.9.0
 API version:  1.21
 Go version:   go1.4.3
 Git commit:   76d6bc9
 Built:        Tue Nov  3 22:16:18 UTC 2015
 OS/Arch:      linux/arm
```

`docker info`:
```
Containers: 1
Images: 48
Server Version: 1.9.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 50
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.2.5-263
Operating System: Ubuntu 15.10
CPUs: 4
Total Memory: 1.974 GiB
Name: scw-05
ID: DM7U:NQYF:UL5H:4ZQY:OTHQ:YZ3A:GH3X:IDVJ:Z6ZL:DIE5:3VJS:MM3G
Labels:
 provider=scaleway
```

`uname -a`:
```
Linux scw-05 4.2.5-263 #1 SMP Wed Nov 4 17:03:17 UTC 2015 armv7l armv7l armv7l GNU/Linux
```


Environment details (AWS, VirtualBox, physical, etc.):
Scaleway C1 ARM physical server

How reproducible:


Steps to Reproduce:
1. Add some debug infos into the integration test TestDockerNetworkHostModeUngracefulDaemonRestart:

```
diff --git a/integration-cli/docker_cli_network_unix_test.go b/integration-cli/docker_cli_network_unix_test.go
index d357e42..a245923 100644
--- a/integration-cli/docker_cli_network_unix_test.go
+++ b/integration-cli/docker_cli_network_unix_test.go
@@ -798,6 +798,10 @@ func (s *DockerNetworkSuite) TestDockerNetworkHostModeUngracefulDaemonRestart(c
        testRequires(c, DaemonIsLinux, NotUserNamespace)
        s.d.Start()
 
+       fmt.Println("######DEBUG#######")
+       imagesOut, _ := s.d.Cmd("images")
+       fmt.Println(imagesOut)
+
        // Run a few containers on host network
        for i := 0; i < 10; i++ {
                cName := fmt.Sprintf("hostc-%d", i)
```

2. Run the `TESTFLAGS='-check.f TestDockerNetworkHostModeUngracefulDaemonRestart' make test-integration-cli` on ARM or any other architecture.
3.


Actual Results:

## Actual results for linux/arm

```
---> Making bundle: .integration-daemon-setup (in bundles/1.10.0-dev/test-integration-cli)
---> Making bundle: .ensure-emptyfs (in bundles/1.10.0-dev/test-integration-cli)
++++ tar -cC bundles/1.10.0-dev/test-integration-cli/emptyfs .
++++ docker load
---> Making bundle: .ensure-frozen-images (in bundles/1.10.0-dev/test-integration-cli)
++++ tar -cC /docker-frozen-images .
++++ docker load
Untagged: hypriot/armhf-busybox:latest
Untagged: hypriot/armhf-hello-world:latest
Untagged: hypriot/armhf-unshare:latest
---> Making bundle: .ensure-images (in bundles/1.10.0-dev/test-integration-cli)
---> Making bundle: .ensure-httpserver (in bundles/1.10.0-dev/test-integration-cli)
+ go test -check.f TestDockerNetworkHostModeUngracefulDaemonRestart -test.timeout=60m -check.v github.com/docker/docker/integration-cli
INFO: Testing against a local daemon
######DEBUG#######
REPOSITORY          TAG                 IMAGE ID            CREATED             VIRTUAL SIZE


----------------------------------------------------------------------
FAIL: docker_cli_network_unix_test.go:797: DockerNetworkSuite.TestDockerNetworkHostModeUngracefulDaemonRestart

[d47470915] waiting for daemon to start
[d47470915] daemon started
docker_cli_network_unix_test.go:809:
    c.Assert(err, checker.IsNil, check.Commentf(out))
... value *exec.ExitError = &exec.ExitError{ProcessState:(*os.ProcessState)(0x10b9aff0)} ("exit status 125")
... Unable to find image 'busybox:latest' locally
latest: Pulling from library/busybox
039b63dd2cba: Pulling fs layer
c51f86c28340: Pulling fs layer
c51f86c28340: Verifying Checksum
c51f86c28340: Download complete
039b63dd2cba: Verifying Checksum
039b63dd2cba: Download complete
039b63dd2cba: Pull complete
c51f86c28340: Pull complete
Digest: sha256:eb3c0d4680f9213ee5f348ea6d39489a1f85a318a2ae09e012c426f78252a6d2
Status: Downloaded newer image for busybox:latest
acfc28eedce5f72b3f6853e2bca6432c557333881d9fc30e0ff50f1f14cc817b
docker: Error response from daemon: Cannot start container acfc28eedce5f72b3f6853e2bca6432c557333881d9fc30e0ff50f1f14cc817b: [8] System error: exec format error.


[d47470915] exiting daemon
OOPS: 0 passed, 1 FAILED
--- FAIL: Test (10.72s)
FAIL
coverage: 20.5% of statements
exit status 1
FAIL	github.com/docker/docker/integration-cli	11.989s
---> Making bundle: .integration-daemon-stop (in bundles/1.10.0-dev/test-integration-cli)
+++++ cat bundles/1.10.0-dev/test-integration-cli/docker.pid
```

## Actual results for linux/amd64
```
---> Making bundle: .integration-daemon-setup (in bundles/1.10.0-dev/test-integration-cli)
---> Making bundle: .ensure-emptyfs (in bundles/1.10.0-dev/test-integration-cli)
++++ docker load
++++ tar -cC bundles/1.10.0-dev/test-integration-cli/emptyfs .
---> Making bundle: .ensure-frozen-images (in bundles/1.10.0-dev/test-integration-cli)
++++ docker load
++++ tar -cC /docker-frozen-images .
---> Making bundle: .ensure-images (in bundles/1.10.0-dev/test-integration-cli)
---> Making bundle: .ensure-httpserver (in bundles/1.10.0-dev/test-integration-cli)
+ go test -check.f TestDockerNetworkHostModeUngracefulDaemonRestart -test.timeout=60m -check.v github.com/docker/docker/integration-cli
INFO: Testing against a local daemon
######DEBUG#######
REPOSITORY          TAG                 IMAGE ID            CREATED             VIRTUAL SIZE

PASS: docker_cli_network_unix_test.go:797: DockerNetworkSuite.TestDockerNetworkHostModeUngracefulDaemonRestart	13.882s
OK: 1 passed
PASS
coverage: 20.7% of statements
ok  	github.com/docker/docker/integration-cli	14.225s
---> Making bundle: .integration-daemon-stop (in bundles/1.10.0-dev/test-integration-cli)
++++ cat bundles/1.10.0-dev/test-integration-cli/docker.pid
```

Expected Results:

The already preloaded images for busybox must be used as it is pinned and loaded before all tests as well as pulled for the correct architecture and tagged as `busybox`.
A direct pull from Docker Hub doesn't seem intended for me here.

Additional info:

The implicit docker pull seems to me the fact that the test takes 10-14 seconds as it depends on internet connectivity.

The test TestDockerNetworkHostModeUngracefulDaemonRestart was introduced with docker/docker#17548


Any hints how to start another docker daemon that also has the images loaded in such integration tests?

Or should we skip these tests for ARM for now?

This is related to the work-in-progress of Support Docker on ARM #17802

--- a/integration-cli/docker_cli_network_unix_test.go
+++ b/integration-cli/docker_cli_network_unix_test.go
@@ -753,7 +753,7 @@ func (s *DockerNetworkSuite) TestDockerNetworkMultipleNetworksGracefulDaemonRest
 	cName := "bb"
 	nwList := []string{"nw1", "nw2", "nw3"}
 
-	s.d.Start()
+	s.d.StartWithBusybox()
 
 	connectContainerToNetworks(c, s.d, cName, nwList)
 	verifyContainerIsConnectedToNetworks(c, s.d, cName, nwList)
@@ -771,7 +771,7 @@ func (s *DockerNetworkSuite) TestDockerNetworkMultipleNetworksUngracefulDaemonRe
 	cName := "cc"
 	nwList := []string{"nw1", "nw2", "nw3"}
 
-	s.d.Start()
+	s.d.StartWithBusybox()
 
 	connectContainerToNetworks(c, s.d, cName, nwList)
 	verifyContainerIsConnectedToNetworks(c, s.d, cName, nwList)
@@ -796,7 +796,7 @@ func (s *DockerNetworkSuite) TestDockerNetworkRunNetByID(c *check.C) {
 
 func (s *DockerNetworkSuite) TestDockerNetworkHostModeUngracefulDaemonRestart(c *check.C) {
 	testRequires(c, DaemonIsLinux, NotUserNamespace)
-	s.d.Start()
+	s.d.StartWithBusybox()
 
 	// Run a few containers on host network
 	for i := 0; i < 10; i++ {

